haystack/preview/components/samplers/top_p.py

```diff
@@ -0,0 +1,142 @@
+import logging
+from typing import List, Optional, Dict, Any
+
+from haystack.preview import ComponentError
+
+from haystack.lazy_imports import LazyImport
+from haystack.preview import Document, component, default_from_dict, default_to_dict
+
+
+logger = logging.getLogger(__name__)
+
+
+with LazyImport(message="Run 'pip install transformers[torch,sentencepiece]==4.32.1'") as torch_and_transformers_import:
+    import torch
+
+
+@component
+class TopPSampler:
+    """
+    Selects documents based on the cumulative probability of the Document similarity scores using top p (nucleus)
+    sampling.
+
+    Usage example:
+    ```
+    from haystack.preview import Document
+    from haystack.preview.components.samplers import TopPSampler
+
+    sampler = TopPSampler(top_p=0.95)
+    docs = [
+        Document(text="Berlin", metadata={"similarity_score": -10.6}),
+        Document(text="Belgrade", metadata={"similarity_score": -8.9}),
+        Document(text="Sarajevo", metadata={"similarity_score": -4.6}),
+    ]
+    query = "City in Bosnia and Herzegovina"
+    output = sampler.run(documents=docs)
+    docs = output["documents"]
+    assert len(docs) == 1
+    assert docs[0].text == "Sarajevo"
+    ```
+    """
+
+    def __init__(self, top_p: Optional[float] = 1.0, score_field: Optional[str] = "similarity_score"):
```

Comment:
Does it make sense for this params to be `Optional`?

Reply:
No, you are right, thank you